src/lib/viz/layer/Layer.ts

```diff
@@ -59,6 +70,8 @@ export class Layer extends WithEvents implements StyledLayer {
     this._style = buildStyle(style);
 
     this.registerAvailableEvents([
+      DATA_READY_EVENT,
+      DATA_CHANGED_EVENT,
       'viewportLoad',
```

Comment:
I like the idea of CONSTANTS for events here, indeed we could also add them for the others